gdnative-core/src/variant_array.rs

```diff
@@ -171,107 +156,374 @@ impl VariantArray {
+        unsafe {
+            let mut sys = sys::godot_array::default();
+            (get_api().godot_array_new)(&mut sys);
+            Self::from_sys(sys)
         }
     }
 
-    #[doc(hidden)]
+    /// Put this array under the "shared" access type.
     #[inline]
-    pub fn sys(&self) -> *const sys::godot_array {
-        &self.0
+    pub fn into_shared(self) -> VariantArray<Shared> {
+        let sys = self.sys;
+        std::mem::forget(self);
+
+        VariantArray::<Shared> {
+            sys,
+            _marker: PhantomData,
+        }
     }
 
-    #[doc(hidden)]
+    /// Clears the array, resizing to 0.
     #[inline]
-    pub fn from_sys(sys: sys::godot_array) -> Self {
-        VariantArray(sys)
+    pub fn clear(&self) {
+        unsafe {
+            (get_api().godot_array_clear)(self.sys_mut());
+        }
+    }
+
+    /// Removes the element at `idx`.
+    #[inline]
+    pub fn remove(&self, idx: i32) {
+        unsafe { (get_api().godot_array_remove)(self.sys_mut(), idx) }
+    }
+
+    /// Removed the first occurrence of `val`.
+    #[inline]
+    pub fn erase(&self, val: &Variant) {
+        unsafe { (get_api().godot_array_erase)(self.sys_mut(), val.sys()) }
+    }
+
+    #[inline]
+    pub fn resize(&self, size: i32) {
+        unsafe { (get_api().godot_array_resize)(self.sys_mut(), size) }
+    }
+
+    /// Appends an element at the end of the array.
+    #[inline]
+    pub fn push(&self, val: &Variant) {
+        unsafe {
+            (get_api().godot_array_push_back)(self.sys_mut(), val.sys());
+        }
+    }
+
+    /// Removes an element at the end of the array.
+    #[inline]
+    pub fn pop(&self) -> Variant {
+        unsafe { Variant((get_api().godot_array_pop_back)(self.sys_mut())) }
+    }
+
+    /// Appends an element to the front of the array.
+    #[inline]
+    pub fn push_front(&self, val: &Variant) {
+        unsafe {
+            (get_api().godot_array_push_front)(self.sys_mut(), val.sys());
+        }
+    }
+
+    /// Removes an element at the front of the array.
+    #[inline]
+    pub fn pop_front(&self) -> Variant {
+        unsafe { Variant((get_api().godot_array_pop_front)(self.sys_mut())) }
+    }
+
+    /// Insert a new int at a given position in the array.
+    #[inline]
+    pub fn insert(&self, at: i32, val: &Variant) {
+        unsafe { (get_api().godot_array_insert)(self.sys_mut(), at, val.sys()) }
+    }
+
+    /// Returns an iterator through all values in the `VariantArray`.
+    ///
+    /// `VariantArray` is reference-counted and have interior mutability in Rust parlance.
+    /// Modifying the same underlying collection while observing the safety assumptions will
+    /// not violate memory safely, but may lead to surprising behavior in the iterator.
+    #[inline]
+    pub fn iter(&self) -> IterUnique {
+        self.into_iter()
+    }
+}
+
+impl VariantArray<Shared> {
+    /// Create a new shared array.
+    #[inline]
+    pub fn new_shared() -> Self {
```

Comment:
This reminds me there is no `new_shared` for `Dictionary`. Perhaps one can be added as well?

Reply:
Added that one to `Dictionary` too, good catch!